lib/workers/repository/update/branch/auto-replace.spec.ts

```diff
@@ -1038,5 +1038,250 @@ describe('workers/repository/update/branch/auto-replace', () => {
         `
       );
     });
+
+    it('docker: updates with pinDigest enabled but no currentDigest value', async () => {
+      const dockerfile = codeBlock`
+        FROM ubuntu:18.04
+      `;
+      upgrade.manager = 'dockerfile';
+      upgrade.depName = 'ubuntu';
+      upgrade.currentValue = '18.04';
+      upgrade.currentDigest = undefined;
+      upgrade.depIndex = 0;
+      upgrade.pinDigests = true;
+      upgrade.updateType = 'replacement';
+      upgrade.replaceString = 'ubuntu:18.04';
+      upgrade.newName = 'alpine';
+      upgrade.newValue = '3.16';
+      upgrade.newDigest = 'sha256:p0o9i8u7z6t5r4e3w2q1';
+      upgrade.packageFile = 'Dockerfile';
+      const res = await doAutoReplace(upgrade, dockerfile, reuseExistingBranch);
+      expect(res).toBe(
+        codeBlock`
+          FROM alpine:3.16
+        `
+      );
+    });
+
+    it('docker: updates with pinDigest enabled and a currentDigest value', async () => {
+      const dockerfile = codeBlock`
+        FROM ubuntu:18.04@sha256:q1w2e3r4t5z6u7i8o9p0
+      `;
+      upgrade.manager = 'dockerfile';
+      upgrade.depName = 'ubuntu';
+      upgrade.currentValue = '18.04';
+      upgrade.currentDigest = 'sha256:q1w2e3r4t5z6u7i8o9p0';
+      upgrade.depIndex = 0;
+      upgrade.pinDigests = true;
+      upgrade.updateType = 'replacement';
+      upgrade.replaceString = 'ubuntu:18.04@sha256:q1w2e3r4t5z6u7i8o9p0';
+      upgrade.newName = 'alpine';
+      upgrade.newValue = '3.16';
+      upgrade.newDigest = 'sha256:p0o9i8u7z6t5r4e3w2q1';
+      upgrade.packageFile = 'Dockerfile';
+      const res = await doAutoReplace(upgrade, dockerfile, reuseExistingBranch);
+      expect(res).toBe(
+        codeBlock`
+          FROM alpine:3.16@sha256:p0o9i8u7z6t5r4e3w2q1
+        `
+      );
+    });
+
+    it('regex: updates with pinDigest enabled but no currentDigest value', async () => {
+      const yml = 'image: "some.url.com/my-repository:1.0"';
+      upgrade.manager = 'regex';
+      upgrade.pinDigests = true;
+      upgrade.depName = 'some.url.com/my-repository';
+      upgrade.currentValue = '1.0';
+      upgrade.currentDigest = undefined;
+      upgrade.depIndex = 0;
+      upgrade.replaceString = 'image: "some.url.com/my-repository:1.0"';
+      upgrade.packageFile = 'k8s/base/defaults.yaml';
+      upgrade.newName = 'some.other.url.com/some-new-repo';
+      upgrade.newValue = '3.16';
+      upgrade.newDigest = 'sha256:p0o9i8u7z6t5r4e3w2q1';
+      upgrade.matchStrings = [
+        'image:\\s*?\\\'?\\"?(?<depName>[^:\\\'\\"]+):(?<currentValue>[^@\\\'\\"]+)@?(?<currentDigest>[^\\s\\\'\\"]+)?\\"?\\\'?\\s*',
+      ];
+      const res = await doAutoReplace(upgrade, yml, reuseExistingBranch);
+      expect(res).toBe('image: "some.other.url.com/some-new-repo:3.16"');
```

Comment:
Same here

Reply:
Please also add an example with a set `autoReplaceStringTemplate` where the digest is added.

Reply:
As above